src/llvm-late-gc-lowering.cpp

```diff
@@ -2320,6 +2321,59 @@ MDNode *createMutableTBAAAccessTag(MDNode *Tag) {
     return MDBuilder(Tag->getContext()).createMutableTBAAAccessTag(Tag);
 }
 
+void LateLowerGCFrame::CleanupWriteBarriers(Function &F, State *S, SmallVector<CallInst*, 0> &WriteBarriers, bool *CFGModified) {
```

Comment:
```suggestion
void LateLowerGCFrame::CleanupWriteBarriers(Function &F, const State *S, const SmallVector<CallInst*, 0> &WriteBarriers, bool *CFGModified) {
```
?

Reply:
`State *S` is passed to a bunch of functions (e.g. `IsPermRooted`) which take a non-const pointer as argument. Other than that making `WriteBarriers` a const reference should be fine.

Thanks for catching it!